### PR TITLE
Bring in reusable theme feature and tests

### PR DESCRIPTION
**Pull request by caalador**
part of #9983

### CODE REVIEW

File: flow-tests/test-embedding/test-embedding-reusable-theme/src/test/java/com/vaadin/flow/webcomponent/ApplicationThemeComponentIT.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import com.vaadin.flow.component.html.testbench.SpanElement;
+import com.vaadin.flow.testutil.ChromeBrowserTest;
+import com.vaadin.testbench.TestBenchElement;
+
+import static com.vaadin.flow.webcomponent.ThemedComponent.EMBEDDED_ID;
+import static com.vaadin.flow.webcomponent.ThemedComponent.HAND_ID;
+import static com.vaadin.flow.webcomponent.ThemedComponent.MY_LIT_ID;
+import static com.vaadin.flow.webcomponent.ThemedComponent.MY_POLYMER_ID;
+
+public class ApplicationThemeComponentIT extends ChromeBrowserTest {
```
Comment (pleku) [inline flow-tests/test-embedding/test-embedding-reusable-theme/src/test/java/com/vaadin/flow/webcomponent/ApplicationThemeComponentIT.java:33]: The working title of the feature, _application theme_, seems to have stayed heavily on the code side but :this-is-fine:

**Review by pleku — COMMENTED** [review]
One question

File: flow-tests/test-embedding/test-embedding-reusable-theme/src/test/java/com/vaadin/flow/webcomponent/ApplicationThemeComponentIT.java (hunk 1 above)
Comment (caalador) [reply flow-tests/test-embedding/test-embedding-reusable-theme/src/test/java/com/vaadin/flow/webcomponent/ApplicationThemeComponentIT.java:33]: Would need to later be updated to the master and then backported to 4.x

File: flow-tests/pom.xml [hunk 2]
```diff
@@ -219,9 +219,6 @@
                         </execution>
                     </executions>
                     <configuration>
-                        <webAppConfig>
-                            <containerIncludeJarPattern>^$</containerIncludeJarPattern>
-                        </webAppConfig>
```
[the remark this answers is not in the document]
Comment (caalador) [reply flow-tests/pom.xml:224]: Could be from some merge. Can return, but like this we don't get hundreds of lines saying some class found in 2 different places.

**Review by pleku — APPROVED** [review]

**Comment by pleku** [comment]
This probably needs to be squashed ? Not all commits in history follow the messaage syntax

**Comment by caalador** [comment]
Well they are as they were in master. 
Should have noted when cherry picking.

**Comment by pleku** [comment]
It could be fixed by f-p but the release notes will need to be cleaned up a bit anyway so I don't think it matters so much

**Pull request merged**